engine/src/gldrv/gl_light_state.cpp

```diff
@@ -154,9 +154,25 @@ int _GLLightsEnabled    = 0;
 Hashtable3d< LineCollideStar, 20, CTACC, lighthuge >lighttable;
 
 GFXLight gfx_light::operator=( const GFXLight &tmp )
-{
-    memcpy( this, &tmp, sizeof (GFXLight) );
-    return tmp;
+{   // Let's see if I can write a better copy operator
+//    memcpy( this, &tmp, sizeof (GFXLight) );
+            this->target  = -1;
+            this->options = 0;
+            memcpy( this->vect, tmp.vect, sizeof (float)*3 );
+            memcpy( this->diffuse, tmp.diffuse, sizeof (float)*4 );
+            memcpy( this->specular, tmp.specular, sizeof (float)*4 );
+            memcpy( this->ambient, tmp.ambient, sizeof (float)*4 );
+            memcpy( this->attenuate, tmp.attenuate, sizeof (float)*3 );
+            memcpy( this->direction, tmp.direction, sizeof (this->direction) );
+            this->exp    = tmp.exp;
+            this->cutoff = tmp.cutoff;
+            this->size   = tmp.size;
```

Comment:
I think you missed `occlusion`

Reply:
Indeed I did. Thanks for picking up on it.